### PR TITLE
Add block inspector virtual bubbling option

### LINKED ISSUE #24809: Navigation screen: Toggling a panel and then selecting an input will hide Block Settings
Split out from https://github.com/WordPress/gutenberg/issues/24760.

1. Go to the Navigation screen.
1. Open Block Settings.
1. Toggle open a panel.
1. Select an input within that panel.

The Block Settings popover is closed. It should remain open.

### PR DESCRIPTION
**Pull request by tellthemachines**
## Description

Fixes #24809 

The block inspector in the Navigation screen is inside a Dropdown, and its default `bubblesVirtually` causes Dropdown's `withFocusOutside` wrapper to not receive focus/blur events correctly. This PR adds a `bubblesVirtually` prop to `BlockInspector`, defaulting to `true`, so that it can be set to `false` whenever the inspector needs to be used inside a dropdown or popover.

## How has this been tested?

Tested in browser, checked keyboard navigation follows the correct sequence.

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/block-inspector/index.js [hunk 1]
```diff
@@ -30,12 +30,17 @@ const BlockInspector = ( {
...
 	if ( count > 1 ) {
 		return (
 			<div className="block-editor-block-inspector">
 				<MultiSelectionInspector />
-				<InspectorControls.Slot bubblesVirtually />
+				{ bubblesVirtually ? (
+					<InspectorControls.Slot bubblesVirtually />
+				) : (
+					<InspectorControls.Slot />
+				) }
```
Comment (talldan) [inline packages/block-editor/src/components/block-inspector/index.js:43]: Could these be written without the ternary?
```suggestion
				<InspectorControls.Slot bubblesVirtually={ bubblesVirtually } />
```

**Review by talldan — APPROVED** [review]
This does the trick. Tested both the post editor and navigation editor. The post editor still behaves the same. 

The block inspector in the nav screen now doesn't close unexpectedly, focus is retained, and `Escape` can still be used to close the popover with focus returning to the opening button.

File: packages/block-editor/src/components/block-inspector/index.js (hunk 1 above)
Comment (tellthemachines) [reply packages/block-editor/src/components/block-inspector/index.js:43]: Oh, yes, of course!

**Comment by youknowriad** [comment]
I think the main issue is that for DOM specific events where we rely on DOM bubbling instead of React event bubbling, we should rely on DOM events `addEventListener` and not React handlers.

In other words, I think the issues is not in the inspector and the inspector should always use bubblesVirtually. The issue instead is in `withFocusOutside` where it should use DOM events and not React events.

**Comment by talldan** [comment]
> The issue instead is in withFocusOutside where it should use DOM events and not React events.

@youknowriad Though `blur` doesn't bubble IIRC, which this component depends on. I wonder if it may also cause some issues with nested popovers if it was implemented with DOM event listeners as blur events would occur outside the originating popover.

**Comment by youknowriad** [comment]
@talldan Can we rely on event capturing maybe?

**Comment by tellthemachines** [comment]
We could give it a try, but there's a risk it might break existing uses of `withFocusOutside` that rely on the event bubbling virtually. 

> I wonder if it may also cause some issues with nested popovers if it was implemented with DOM event listeners as blur events would occur outside the originating popover.

Don't we already have some issues with nested popovers/modals not closing when expected? Maybe worth checking if this would actually fix them 😅

**Comment by youknowriad** [comment]
> there's a risk it might break existing uses of withFocusOutside that rely on the event bubbling virtually.

conceptually, it seems that withFocusOutside shouldn't rely on React event bubbling anyway. I'm having a hard time thinking about any potential use-case there. Doesn't mean it doesn't exist though.

**Comment by youknowriad** [comment]
@talldan Can we try that on a PR and see how it looks?

**Comment by youknowriad** [comment]
I didn't mention but omitting event bubbling in the inspector breaks the inspector for folks using RichText on the inspector.

**Pull request merged**